### DISCUSSION TITLE
Change device class of sensor entity via “Show as” selector

### DISCUSSION THREAD

**Discussion by krazos** [home-assistant/frontend#13673] [Configuration]
You can change the `device_class` of a binary sensor via the “Show as” selector in the UI, but this functionality is not available for (non-binary) sensors. It would be great to have similar functionality for the sensor domain.

Example: I use the min/max/mean helper to create average temperature measurements from multiple temperature sensors. However, the min/max/mean sensor does not inherit a `device_class` or icon from the constituent sensors, and the only way to set the appropriate `device_class` is through YAML.

See also: https://github.com/home-assistant/core/issues/76003

**Comment by tvwerkhoven** [comment]
For reference, the YAML solution is the following, see also [this related thread](https://github.com/home-assistant/frontend/issues/19474).

```yaml
homeassistant:
  customize:
    sensor.outdoor_temperature:
      device_class: temperature
```
